### PR TITLE
DL: Standardize the data types used for keras and sql

### PR DESCRIPTION
**Pull request by jingyimei**
JIRA: MADLIB-1358

Under deep_learning module (this includes preprocessor, keras model arch
table, fit, evaluate and predict), we made the following changes:

1. independent_var and normalizing const to be float32 (REAL).
2. dependent_var to be a 16-bit integer (SMALLINT)

We were previously converting independent_var to float64 which is not
necessary at all. Instead we might be able to save resources by just
using float32 for the independent_var and int16 for the one hot encoded
dependent_var.

Co-authored-by: Nikhil Kak <nkak@pivotal.io>

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/978/

File: src/ports/postgres/modules/deep_learning/madlib_keras_helper.py_in [hunk 1]
```diff
@@ -160,3 +164,7 @@ METRIC_TYPE_COLNAME = "metrics_type"
 # MADLIB-1300 Adding these variables for DL only at this time.
 MINIBATCH_OUTPUT_DEPENDENT_COLNAME_DL = "dependent_var"
 MINIBATCH_OUTPUT_INDEPENDENT_COLNAME_DL = "independent_var"
+
+## sql variable types
```
Comment (orhankislal) [inline src/ports/postgres/modules/deep_learning/madlib_keras_helper.py_in]: Maybe we should define these constants at the top of the file.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/981/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/982/

**Pull request merged**